### ISSUE TITLE
Passing strings through the iterator without conversion to a tensor

### ISSUE THREAD

**Issue by martiansideofthemoon** [allenai/allennlp#2789]
If I understand correctly, the AllenNLP model uses the functions in a `Field` object to convert each type of data in an `Instance` into a numeric tensor which can be batched / padded by the iterator.

Is there a `Field` object which bypasses the "numeric tensor" restriction? The iterator would ignore indexing as well as padding these special `Field`objects and simply return a list of strings for the batch (similar to how Pytorch's `DataLoader` would work) instead of a tensor of padded + indexed strings.

If no such `Field` object exists, is there any other way of doing this? I am currently doing this in a hacky way by passing these strings as `LabelField` objects, setting `shuffle=False` in the iterator and looking up the dataset object (the list of `Instance`) to procure the original un-indexed `LabelField`.

This will be useful during debugging, especially when we are using the iterator objects in custom for loops.

**Comment by joelgrus** [comment]
there is a `MetadataField` that can contain anything and just hands it to the model as-is

https://allenai.github.io/allennlp-docs/api/allennlp.data.fields.html#metadata-field
